modules/gapi/test/infer/gapi_infer_ov_tests.cpp

```diff
@@ -657,6 +658,49 @@ TEST_F(TestAgeGenderListOV, InferList2Generic_Image) {
     validate();
 }
 
+TEST(TestOV, Infer_ImageCorrectNHWCInputLayout) {
+    const std::string model_name   = "ModelNHWC";
+    const std::string model_path   = model_name + ".xml";
+    const std::string weights_path = model_name + ".bin";
+    const std::string device_id    = "CPU";
+    const int W                    = 128;
+    const int H                    = 64;
+    const int C                    = 3;
+    const int N                    = 1;
+
+    auto data1 = std::make_shared<ov::opset8::Parameter>(ov::element::u8, ov::Shape{N, H, W, C});
+    data1->output(0).set_names({"data1_t"});  // tensor names
+    ov::layout::set_layout(data1, ov::Layout("NHWC"));
+
+    auto sin = std::make_shared<ov::opset8::Sin>(data1);
+    sin->output(0).set_names({"sin_t"});  // tensor name
+
+    auto result = std::make_shared<ov::opset8::Result>(sin);
+    result->output(0).set_names({"result_t"});  // tensor name
+
+    auto f = std::make_shared<ov::Model>(ov::ResultVector{result}, ov::ParameterVector{data1}, "function_name");
+
+    ov::serialize(f, model_path, weights_path, ov::pass::Serialize::Version::IR_V11);
```

Comment:
Probably there's some general approach to producing temporary files in tests in OpenCV.. Like some specific folder or naming convention?

Reply:
btw maybe we in the future we need to extend our API to accept objects of `ov::Model` directly instead of xml/bin

Reply:
`plaidml.hpp` looks clear from framework details